src/main/java/madstax/view/PlannerCellRenderer.java

```diff
@@ -26,9 +26,30 @@ public Component getTableCellRendererComponent(JTable table, Object value, boole
             default:
                 s = value.toString();
         }
+
         Component c = renderer.getTableCellRendererComponent(table, s,
                 isSelected, hasFocus, row, col);
-        ((JLabel) c).setHorizontalAlignment(SwingConstants.LEFT);
+        ((JLabel) c).setHorizontalAlignment(SwingConstants.CENTER);
+
+        if(col == 4){
+            switch(s){
+                case "Approved":
```

Comment:
This defeats the purpose of the `RequestStatus` enumeration because you are actually comparing the display string. One should also be careful while doing it! String comparison should be done through the `.equals()` method. This is a very expensive operation.

https://www.geeksforgeeks.org/string-in-switch-case-in-java/

A better approach would be to manipulate the `RequestStatus` enumeration itself.

Reply:
Yay I actually understood this and managed to fix it all by myself XD